### PR TITLE
Add imminent events from more sources

### PR DESCRIPTION
**Pull request by frozenhelium**

### CODE REVIEW

File: src/utils/risk.ts [hunk 1]
```diff
@@ -325,6 +325,31 @@ export function isValidFeature(
...
+    const safeObj = maybeFeatureCollection as GeoJSON.FeatureCollection;
+    if (safeObj.type !== 'FeatureCollection') {
+        return false;
+    }
+
+    if (!Array.isArray(safeObj.features)) {
+        return false;
+    }
+
+    // TODO: validate each feature?
```
Comment (tnagorra) [inline src/utils/risk.ts:348]: We can use a library like geojsonhint
https://github.com/mapbox/geojsonhint

And then validate only the properties

File: src/hooks/useCountry.ts [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+    id?: never;
+    region?: never;
+}
+
+function useCountry(props?: ListProps): Array<Country>
+function useCountry(props: PropsForId): Country | undefined
+function useCountry(props: PropsForIso3): Country | undefined
+function useCountry(
+    props?: ListProps | PropsForId | PropsForIso3,
+): (Country | undefined | Array<Country>) {
```
Comment (tnagorra) [inline src/hooks/useCountry.ts:47]: It's a bit weird to return both array and a single country

File: src/hooks/useCountry.ts [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+                setPending(true);
+
+                return () => {
+                    setPending(false);
+                };
+            }
+
+            return undefined;
+        },
+        [countries, pending, setPending],
```
Comment (tnagorra) [inline src/hooks/useCountry.ts]: Why do we need this useEffect?
Why not just use pending from useRequest

File: src/hooks/useCountry.ts [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+                    return undefined;
+                }
+
+                return {
+                    ...country,
+                    id: country.id,
+                    name: country.name,
+                    iso: country.iso,
+                    iso3: country.iso3,
+                };
```
Comment (tnagorra) [inline src/hooks/useCountry.ts]: Is this something we need to make a utility?

File: src/components/RiskImminentEvents/index.tsx [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+    {
+        key: 'DR',
+        label: 'Drought',
+        icon: <DroughtIcon className={styles.icon} />,
+    },
+    {
+        key: 'WF',
+        label: 'Wildfire',
+        icon: null,
+    },
```
Comment (tnagorra) [inline src/components/RiskImminentEvents/index.tsx:62]: Don't we need to translate the label? Don't we have a enum query for hazard?

**Review by tnagorra — CHANGES_REQUESTED** [review]

**Review by tnagorra — COMMENTED** [review]
Let's fix all of the comments before we can merge

File: src/utils/common.ts [hunk 6]
```diff
@@ -27,30 +28,48 @@ function getNumberListSafe(list: UnsafeNumberList) {
...
+export function minSafe(list: UnsafeNumberList | NumberList) {
     const safeList = getNumberListSafe(list);
-    if (!safeList || safeList.length === 0) {
+    if (!safeList) {
         return undefined;
     }
 
+    if (safeList.length === 0) {
+        return 0;
+    }
```
Comment (tnagorra) [inline src/utils/common.ts]: It's not correct to return 0 as min value when the list is empty

File: src/utils/common.ts [hunk 7]
```diff
@@ -27,30 +28,48 @@ function getNumberListSafe(list: UnsafeNumberList) {
...
+export function maxSafe(list: NumberList | UnsafeNumberList) {
     const safeList = getNumberListSafe(list);
-    if (!safeList || safeList.length === 0) {
+    if (!safeList) {
         return undefined;
     }
 
+    if (safeList.length === 0) {
+        return 0;
+    }
```
Comment (tnagorra) [inline src/utils/common.ts]: It's not correct to return 0 as max value when the list is empty

File: src/utils/common.ts [hunk 8]
```diff
@@ -27,30 +28,48 @@ function getNumberListSafe(list: UnsafeNumberList) {
...
+export function sumSafe(list: UnsafeNumberList | NumberList) {
     const safeList = getNumberListSafe(list);
-    if (!safeList || safeList.length === 0) {
+    if (!safeList) {
         return undefined;
     }
 
+    if (safeList.length === 0) {
+        return 0;
+    }
```
Comment (tnagorra) [inline src/utils/common.ts]: It's not correct to return 0 as sum value when the list is empty

File: src/views/RegionSeasonalRiskWatch/Filters/index.tsx [hunk 9]
```diff
@@ -0,0 +1,140 @@
...
+                        label: date.toLocaleString(
+                            navigator.language,
+                            { month: 'long' },
+                        ),
+                    };
+                },
+            );
+        },
+        [],
+    );
```
Comment (tnagorra) [inline src/views/RegionSeasonalRiskWatch/Filters/index.tsx]: We can move this to common

File: src/views/RegionSeasonalRiskWatch/Filters/index.tsx [hunk 10]
```diff
@@ -0,0 +1,140 @@
...
+    const handleChange = useCallback(
+        (
+            newValue: string | string[] | HazardType[] | number[] | boolean,
+            name: 'countries' | 'riskMetric' | 'hazardTypes' | 'months' | 'normalize' | 'includeCopingCapacity',
+        ) => {
+            onChange((prevValue) => ({
+                ...prevValue,
+                [name]: newValue,
+            }));
```
Comment (tnagorra) [inline src/views/RegionSeasonalRiskWatch/Filters/index.tsx]: This is not typesafe.

File: src/views/RegionSeasonalRiskWatch/index.tsx [hunk 11]
```diff
@@ -0,0 +1,629 @@
...
+import type { FilterValue } from './Filters';
+import styles from './styles.module.css';
+
+// eslint-disable-next-line import/prefer-default-export
+export function Component() {
+    const { regionResponse } = useOutletContext<RegionOutletContext>();
+    const bbox = useMemo(
+        () => (regionResponse ? getBbox(regionResponse.bbox) : undefined),
+        [regionResponse],
+    );
```
Comment (tnagorra) [inline src/views/RegionSeasonalRiskWatch/index.tsx]: Why do we use getBbox over bbox to get bbox?

File: src/views/RegionSeasonalRiskWatch/index.tsx [hunk 12]
```diff
@@ -0,0 +1,629 @@
...
+        skip: isNotDefined(regionId),
+        apiType: 'risk',
+        url: '/api/v1/risk-score/',
+        query: {
+            region: isDefined(regionId) ? Number(regionId) : undefined,
+            limit: 500,
+        },
+    });
+
+    const seasonalRiskData = seasonalResponse?.[0];
```
Comment (tnagorra) [inline src/views/RegionSeasonalRiskWatch/index.tsx:96]: Why do we use first elemeent?

File: src/views/RegionSeasonalRiskWatch/index.tsx [hunk 13]
```diff
@@ -0,0 +1,629 @@
...
+                            ...item,
+                            lcc: item.lcc,
+                            country_details: {
+                                ...item.country_details,
+                                iso3: item.country_details.iso3,
+                            },
+                        };
+                    },
+                ).filter(isDefined),
+                (item) => item.country_details.iso3 ?? '',
```
Comment (tnagorra) [inline src/views/RegionSeasonalRiskWatch/index.tsx]:
```suggestion
                (item) => item.country_details.iso3,
```

File: src/views/RegionSeasonalRiskWatch/index.tsx [hunk 14]
```diff
@@ -0,0 +1,629 @@
...
+                (item) => item.lcc,
+            );
+
+            const population = listToMap(
+                riskScoreResponse.results,
+                (item) => item.country_details.iso3 ?? '',
```
Comment (tnagorra) [inline src/views/RegionSeasonalRiskWatch/index.tsx]: Let's filter out the list if there is not iso3

File: src/views/RegionSeasonalRiskWatch/index.tsx [hunk 15]
```diff
@@ -0,0 +1,629 @@
...
+                applicableHazards: {
+                    ...defaultApplicableHazards,
+                    TC: true,
+                    FL: true,
+                    SS: true,
+                },
+            },
+            {
+                key: 'riskScore',
+                label: 'Risk Score',
```
Comment (tnagorra) [inline src/views/RegionSeasonalRiskWatch/index.tsx]: Let's translate these strings

File: src/views/RegionSeasonalRiskWatch/index.tsx [hunk 16]
```diff
@@ -0,0 +1,629 @@
...
+                hazard_type_display: combinedData.hazard_type_display,
+            })).filter(
+                ({ hazard_type: hazard }) => {
+                    if (!selectedRiskMetricDetail?.applicableHazards[hazard]
+                        || !availableHazards[selectedRiskMetricDetail.key][hazard]
+                    ) {
+                        return false;
+                    }
+
+                    return true;
```
Comment (tnagorra) [inline src/views/RegionSeasonalRiskWatch/index.tsx]: We can just return the expression on if statement

File: src/views/RegionSeasonalRiskWatch/index.tsx [hunk 17]
```diff
@@ -0,0 +1,629 @@
...
+                                ...item,
+                                country_details: {
+                                    id: country_details.id,
+                                    iso3: country_details.iso3,
+                                    name: country_details.name,
+                                },
+                            };
+                        },
+                    ).filter(isDefined) ?? [],
+                    (item) => item.country_details.iso3 ?? '',
```
Comment (tnagorra) [inline src/views/RegionSeasonalRiskWatch/index.tsx]:
```suggestion
                    (item) => item.country_details.iso3,
```

File: src/views/RegionSeasonalRiskWatch/index.tsx [hunk 18]
```diff
@@ -0,0 +1,629 @@
...
+                    },
+                ).sort((a, b) => compareNumber(a.totalValue, b.totalValue, -1));
+            }
+
+            return undefined;
+        },
+        [data, filters, mappings],
+    );
+
+    const layerOptions = useMemo<Omit<FillLayer, 'id'>>(
```
Comment (tnagorra) [inline src/views/RegionSeasonalRiskWatch/index.tsx:534]: Let's add a NOTE why we are generate this layer paint

File: src/views/RegionSeasonalRiskWatch/index.tsx [hunk 19]
```diff
@@ -0,0 +1,629 @@
...
+
+                        if (isNotDefined(maxValue) || isNotDefined(sum)) {
+                            return undefined;
+                        }
+
+                        const totalValue = sum;
+                        const normalizedValueListByHazard = valueListByHazard.map(
+                            (item) => ({
+                                ...item,
+                                normalizedValue: item.value / totalValue,
```
Comment (tnagorra) [inline src/views/RegionSeasonalRiskWatch/index.tsx:436]: We should also check if the totalValue is zero

File: src/views/RegionSeasonalRiskWatch/index.tsx [hunk 20]
```diff
@@ -0,0 +1,629 @@
...
+
+                const maxValue = maxSafe(transformedList.map((item) => item.totalValue));
+                if (isNotDefined(maxValue)) {
+                    return undefined;
+                }
+
+                return transformedList.map(
+                    (item) => ({
+                        ...item,
+                        normalizedValue: item.totalValue / maxValue,
```
Comment (tnagorra) [inline src/views/RegionSeasonalRiskWatch/index.tsx:458]: We should check if maxValue is zero

File: src/views/RegionSeasonalRiskWatch/index.tsx [hunk 21]
```diff
@@ -0,0 +1,629 @@
...
+                        if (filters.normalizeByPopulation) {
+                            const population = mappings?.population[item.iso3] ?? 0;
+                            newNormalizedValue *= (population / maxPopulation);
+                            newTotalValue *= (population / maxPopulation);
+                        }
+
+                        if (filters.includeCopingCapacity) {
+                            const lcc = mappings?.lcc[item.iso3] ?? 0;
+                            newNormalizedValue *= (lcc / maxLackOfCopingCapacity);
+                            newTotalValue *= (lcc / maxLackOfCopingCapacity);
```
Comment (tnagorra) [inline src/views/RegionSeasonalRiskWatch/index.tsx]: We should check if these divisors are zero

**Review by tnagorra — CHANGES_REQUESTED** [review]

File: src/views/RegionSeasonalRiskWatch/index.tsx (hunk 11 above)
Comment (tnagorra) [reply src/views/RegionSeasonalRiskWatch/index.tsx]: The type is also being resolved as 'any'

File: src/views/RegionSeasonalRiskWatch/index.tsx (hunk 12 above)
Comment (tnagorra) [reply src/views/RegionSeasonalRiskWatch/index.tsx:96]: There is also some typing issue with the api for seasonal Response

File: src/views/RegionSeasonalRiskWatch/index.tsx (hunk 11 above)
Comment (frozenhelium) [reply src/views/RegionSeasonalRiskWatch/index.tsx]: bbox from server is not compatible with the bbox in mapbox. We need to do `yarn install` or `yarn patch-package` for the types to be resolved properly

File: src/views/RegionSeasonalRiskWatch/index.tsx (hunk 13 above)
Comment (frozenhelium) [reply src/views/RegionSeasonalRiskWatch/index.tsx]: Doesn't matter, we're just creating a look-up map here and `undefined` is not supported as key

File: src/views/RegionSeasonalRiskWatch/index.tsx (hunk 14 above)
Comment (frozenhelium) [reply src/views/RegionSeasonalRiskWatch/index.tsx]: It's a look-up map, so it shouldn't matter

File: src/views/RegionSeasonalRiskWatch/index.tsx (hunk 17 above)
Comment (frozenhelium) [reply src/views/RegionSeasonalRiskWatch/index.tsx]: Doesn't matter for look-up map

Comment (tnagorra) [reply src/views/RegionSeasonalRiskWatch/index.tsx]: It is always defined.

**Pull request merged**